UefiPayloadPkg/Library/PciSegmentInfoLibAcpiBoardInfo/PciSegmentInfoLibAcpiBoardInfo.c

```diff
@@ -55,20 +57,45 @@ RetrieveMultiSegmentInfoFromHob (
   mPciSegments = (PCI_SEGMENT_INFO *)AllocatePool (Size);
   ASSERT (mPciSegments != NULL);
   ZeroMem (mPciSegments, PciRootBridgeInfo->Count * sizeof (PCI_SEGMENT_INFO));
+  pPciSegments = (PCI_SEGMENT_INFO *)AllocatePool (Size);
+  ASSERT (pPciSegments != NULL);
+  ZeroMem (pPciSegments, PciRootBridgeInfo->Count * sizeof (PCI_SEGMENT_INFO));
 
   //
   // Create all root bridges with PciRootBridgeInfoHob
   //
   for (Index = 0; Index < PciRootBridgeInfo->Count; Index++) {
     if (UplSegmentInfo->SegmentInfo[Index].SegmentNumber == (UINT16)(PciRootBridgeInfo->RootBridge[Index].Segment)) {
-      mPciSegments[Index].BaseAddress = UplSegmentInfo->SegmentInfo[Index].BaseAddress;
+      pPciSegments[Index].BaseAddress = UplSegmentInfo->SegmentInfo[Index].BaseAddress;
     }
 
-    mPciSegments[Index].SegmentNumber  = (UINT16)(PciRootBridgeInfo->RootBridge[Index].Segment);
-    mPciSegments[Index].StartBusNumber = (UINT8)PciRootBridgeInfo->RootBridge[Index].Bus.Base;
-    mPciSegments[Index].EndBusNumber   = (UINT8)PciRootBridgeInfo->RootBridge[Index].Bus.Limit;
+    pPciSegments[Index].SegmentNumber  = (UINT16)(PciRootBridgeInfo->RootBridge[Index].Segment);
+    pPciSegments[Index].StartBusNumber = (UINT8)PciRootBridgeInfo->RootBridge[Index].Bus.Base;
+    pPciSegments[Index].EndBusNumber   = (UINT8)PciRootBridgeInfo->RootBridge[Index].Bus.Limit;
+
+    DEBUG ((DEBUG_INFO, "PciRootBridgeInfo->RootBridge[Index].Segment %x\n", PciRootBridgeInfo->RootBridge[Index].Segment));
+    DEBUG ((DEBUG_INFO, "UplSegmentInfo->SegmentInfo[Index].BaseAddress; %x\n", UplSegmentInfo->SegmentInfo[Index].BaseAddress));
+    DEBUG ((DEBUG_INFO, "PciRootBridgeInfo->RootBridge[Index].Bus.Base %x\n", PciRootBridgeInfo->RootBridge[Index].Bus.Base));
+    DEBUG ((DEBUG_INFO, "PciRootBridgeInfo->RootBridge[Index].Bus.Limit %x\n", PciRootBridgeInfo->RootBridge[Index].Bus.Limit));
   }
 
+  Index2 = 0;
+  CopyMem (&mPciSegments[0], &pPciSegments[0], sizeof (PCI_SEGMENT_INFO));
+
+  for (Index = 0; Index < PciRootBridgeInfo->Count; Index++) {

```

Comment:
Index could start with 1 since index 0 was copied in line 83.

Reply:
updated.